### PR TITLE
Make sure we take overflow visible into account when calculating intersections

### PR DESCRIPTION
**Pull request by somewhatabstract**
This addresses an issue we found when integrating the `Tooltip` component with webapp. The bounds checking used to detect visibility did not take into account that regardless of parent element bounds, the anchor may still be visible if the parent has `overflow` set to `visible` for the relevant axis.

This fixes that. We also update all the tests to make sure they have overflow styles to look at and we add a new test case for this new behaviour.

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit a4c6a58152ef41f3f237f1bbc5447daedb052e03

https://deploy-preview-288--wonder-blocks.netlify.com

File: packages/wonder-blocks-core/util/get-element-intersection.js [hunk 1]
```diff
@@ -66,19 +66,29 @@ function getElementIntersectionAgainstParent(
     intersectingRect: ClientRect | DOMRect,
     boundsElement: Element,
 ): Intersection {
+    // We need to check that it matters if we're out of bounds. If the parent
+    // element isn't clipping or otherwise hiding things outside its bounds,
+    // then checking against bounds isn't going to be much use.
+    // So, let's get the style for the element and use the overflow values.
+    const style =
+        ((boundsElement: any).currentStyle: ?CSSStyleDeclaration) ||
+        window.getComputedStyle(boundsElement);
```
Comment (kevinbarabash) [inline packages/wonder-blocks-core/util/get-element-intersection.js]: TIL about `currentStyle`.  https://developer.mozilla.org/en-US/docs/Web/API/Element/currentStyle says it isn't part of any spec... I guess it doesn't matter since we also have a fallback.

File: packages/wonder-blocks-core/util/get-element-intersection.test.js [hunk 2]
```diff
@@ -79,7 +79,13 @@ describe("getElementIntersection", () => {
             // First, let's fake out some scroll parent.
             // Flow doesn't like jest mocks $FlowFixMe
             enumerateScrollAncestors.mockImplementation(() => [
-                {getBoundingClientRect: jest.fn(() => rect(0, 100, 100, 0))},
+                {
+                    getBoundingClientRect: jest.fn(() => rect(0, 100, 100, 0)),
+                    currentStyle: {
+                        overflowX: "auto",
+                        overflowY: "auto",
+                    },
```
Comment (kevinbarabash) [inline packages/wonder-blocks-core/util/get-element-intersection.test.js]: I see why you used `currentStyle`. 🙂

File: packages/wonder-blocks-core/util/get-element-intersection.js (hunk 1 above)
Comment (somewhatabstract) [reply packages/wonder-blocks-core/util/get-element-intersection.js]: Yeah, from my understanding, it's quicker than `getComputedStyle` when it's available, so things often use it just in case.

File: packages/wonder-blocks-core/util/get-element-intersection.test.js (hunk 2 above)
Comment (somewhatabstract) [reply packages/wonder-blocks-core/util/get-element-intersection.test.js]: And yes, it also makes testing slightly easier (wouldn't be hard to have a mock for `getComputedStyle` though...and perhaps we should have a path that tests that.

Comment (kevinbarabash) [reply packages/wonder-blocks-core/util/get-element-intersection.test.js]: Nah, it's cool... you just be mocking it to run the same values.
